src/Raven.Server/Documents/Queries/AST/Query.cs

```diff
@@ -720,7 +720,7 @@ private void SegmentOnRollup(Span<StatefulTimestampValue> values)
                 }
 
                 val = values[index + (int)AggregationType.Count];
-                _count[i] += val.Count;
+                _count[i] += (long)val.Sum;
```

Comment:
Why are you doing this? Shouldn't we worry about losing fractions?

Reply:
This is `Sum` of the `Count`.
To count the total number of points in a rollup segment we need to use the `Sum` of the `Count` value.

Reply:
But you are summing the `Sum`, not ` Count`, am I missing something?

Reply:
In a rollup we have 6 values (First, Last, ... Count)
each of them has `StatefulTimestampValue`, in order to calculate the total points of the original TS for the given rollup segment, we need to go the Count's `StatefulTimestampValue` and use the `Sum` property.

Reply:
ok